src/interpreter/can_open_interpreter.py

```diff
@@ -34,34 +38,66 @@ def inform_interpreter(self, sdo_value, sdo_name, node_properties, index,
             Returns void.
         '''
         self._interpret_object(sdo_value, sdo_name,
+                               sdo_data_type,
                                node_properties['name'],
                                node_properties['type'],
                                index, sub_index)
 
-    def _interpret_object(self, value, name, node_name, node_type_index, index,
-                          sub_index):
+    def _interpret_object(self, value, name, sdo_data_type, node_name,
+                          node_type_index, index, sub_index):
         node_type = NodeType(node_type_index)
+        unpack_format = self._get_unpack_format(sdo_data_type)
         # Checks every possible node
         if(NodeType.DistanceNode == node_type):
             # Adding index, sub_index as additional attributes.
             n_input = self.node_input_factory.create_distance_node_input(
-                unpack('h', value)[0], name, node_name, index=index,
-                sub_index=sub_index)
+                unpack(unpack_format, value)[0], name, node_name, index=index,
+                sub_index=sub_index, data_type=sdo_data_type)
         elif(NodeType.SteeringNode == node_type):
             n_input = self.node_input_factory.create_steering_node_input(
-                unpack('h', value)[0], name, node_name, index=index,
-                sub_index=sub_index)
+                unpack(unpack_format, value)[0], name, node_name, index=index,
+                sub_index=sub_index, data_type=sdo_data_type)
         elif(NodeType.LocalizationNode == node_type):
             n_input = self.node_input_factory.create_localization_node_input(
                 value, name, node_name, index=index,
-                sub_index=sub_index)
+                sub_index=sub_index, data_type=sdo_data_type)
         elif(NodeType.EngineNode == node_type):
-            value_to_bool = True if unpack('h', value)[0] == 1 else False
+            value_to_bool = True if (
+                unpack(unpack_format, value)[0]) == 1 else False
             n_input = self.node_input_factory.create_engine_node_input(
                 value_to_bool, name, node_name, index=index,
                 sub_index=sub_index)
         else:
             n_input = self.node_input_factory.create_temperature_node_input(
-                unpack('h', value)[0], name, node_name, index=index,
-                sub_index=sub_index)
+                unpack(unpack_format, value)[0], name, node_name, index=index,
+                sub_index=sub_index, data_type=sdo_data_type)
         self.vehicle.edit_vehicle_state(n_input)
+
+    def _get_unpack_format(self, can_open_data_type):
```

Comment:
Je kan ook een dictionary gebruiken denk ik

Reply:
Jup daar heb je gelijk in

Reply:
aangepast